tests/test_icdar_format.py

```diff
@@ -0,0 +1,176 @@
+import os.path as osp
+from unittest import TestCase
+
+import numpy as np
+from datumaro.components.extractor import (Bbox, Caption, DatasetItem, Mask,
+    Polygon)
+from datumaro.components.project import Dataset
+from datumaro.plugins.icdar_format.converter import (
+    IcdarTextLocalizationConverter, IcdarTextSegmentationConverter,
+    IcdarWordRecognitionConverter)
+from datumaro.plugins.icdar_format.extractor import IcdarImporter
+from datumaro.util.test_utils import (TestDir, compare_datasets,
+    test_save_and_load)
+
+DUMMY_DATASET_DIR = osp.join(osp.dirname(__file__), 'assets', 'icdar_dataset')
+
+class IcdarImporterTest(TestCase):
+    def test_can_detect(self):
+        self.assertTrue(IcdarImporter.detect(
+            osp.join(DUMMY_DATASET_DIR, 'word_recognition')))
+
+    def test_can_import_captions(self):
+        expected_dataset = Dataset.from_iterable([
+            DatasetItem(id='word_1', subset='train',
+                image=np.ones((10, 15, 3)),
+                annotations=[
+                    Caption('PROPER'),
+                ]
+            ),
+            DatasetItem(id='word_2', subset='train',
+                image=np.ones((10, 15, 3)),
+                annotations=[
+                    Caption("Canon"),
+                ]
+            ),
+        ])
+
+        dataset = Dataset.import_from(osp.join(DUMMY_DATASET_DIR, 'word_recognition'), 'icdar')
+
+        compare_datasets(self, expected_dataset, dataset)
+
+    def test_can_import_bboxes(self):
+        expected_dataset = Dataset.from_iterable([
+            DatasetItem(id='img_1', subset='train',
+                image=np.ones((10, 15, 3)),
+                annotations=[
+                    Polygon([0, 0, 3, 1, 4, 6, 1, 7],
+                        attributes={'text': 'FOOD'}),
+                ]
+            ),
+            DatasetItem(id='img_2', subset='train',
+                image=np.ones((10, 15, 3)),
+                annotations=[
+                    Bbox(0, 0, 2, 3, attributes={'text': 'RED'}),
+                    Bbox(3, 3, 2, 3, attributes={'text': 'LION'}),
+                ]
+            ),
+        ])
+
+        dataset = Dataset.import_from(osp.join(DUMMY_DATASET_DIR, 'text_localization'), 'icdar')
+
+        compare_datasets(self, expected_dataset, dataset)
+
+    def test_can_import_masks(self):
+        expected_dataset = Dataset.from_iterable([
+            DatasetItem(id='1', subset='train',
+                image=np.ones((2, 5, 3)),
+                annotations=[
+                    Mask(id=0, image=np.array([[0, 1, 1, 0, 0], [0, 0, 0, 0, 0]]), label=1,
```

Comment:
What is the point in `center` attribute? Does it help with anything? It can be easily computed from the mask as `mask.get_bbox()` etc.

If any annotation has `label` property set, it should be an index from `LabelCategories`.

Reply:
> What is the point in `center` attribute? Does it help with anything? It can be easily computed from the mask as `mask.get_bbox()` etc.
I tried to find the center by `bbox`, but this does not always work on data of this format. Maybe the center of the symbol is not always the center of the `bbox`.